### DISCUSSION TITLE
fooocus on cpu

### DISCUSSION THREAD

**Discussion by werruww** [lllyasviel/Fooocus#3653] [Q&A]
fooocus run with me on cpu

!pip install tensorflow-cpu

!pip install pygit2==1.15.1
%cd /content
!git clone https://github.com/lllyasviel/Fooocus.git
%cd /content/Fooocus

!python /content/Fooocus/entry_with_update.py --share --always-cpu 

Already up-to-date Update succeeded. [System ARGV] ['[/content/Fooocus/entry_with_update.py](https://colab.research.google.com/drive/1xwBjrCWtZvh-XlTIk2lyQ5sFjDxkcsqW#)', '--share', '--always-cpu'] Python 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] Fooocus version: 2.5.5 [Cleanup] Attempting to delete content of temp dir [/tmp/fooocus](https://colab.research.google.com/drive/1xwBjrCWtZvh-XlTIk2lyQ5sFjDxkcsqW#) [Cleanup] Cleanup successful Running on 48 CPU threads Total VRAM 342591 MB, total RAM 342591 MB Set vram state to: DISABLED Always offload VRAM Device: cpu VAE dtype: torch.float32 [/usr/local/lib/python3.10/dist-packages/numba/np/ufunc/parallel.py](https://colab.research.google.com/drive/1xwBjrCWtZvh-XlTIk2lyQ5sFjDxkcsqW#):371: NumbaWarning: The TBB threading layer requires TBB version 2021 update 6 or later i.e., TBB_INTERFACE_VERSION >= 12060. Found TBB_INTERFACE_VERSION = 12050. The TBB threading layer is disabled. warnings.warn(problem) Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --attention-split Refiner unloaded. Running on local URL: [http://127.0.0.1:7865](https://www.google.com/url?q=http%3A%2F%2F127.0.0.1%3A7865)

IMPORTANT: You are using gradio version 3.41.2, however version 4.29.0 is available, please upgrade.
model_type EPS UNet ADM Dimension 2816 Running on public URL: [https://1b6636ec14a73f6262.gradio.live](https://www.google.com/url?q=https%3A%2F%2F1b6636ec14a73f6262.gradio.live)

This share link expires in 72 hours. For free permanent hosting and GPU upgrades, run gradio deploy from Terminal to deploy to Spaces ([https://huggingface.co/spaces](https://www.google.com/url?q=https%3A%2F%2Fhuggingface.co%2Fspaces)) Using split attention in VAE Working with z of shape (1, 4, 32, 32) = 4096 dimensions. Using split attention in VAE extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_l.logit_scale'} left over keys: dict_keys(['cond_stage_model.clip_l.transformer.text_model.embeddings.position_ids']) Base model loaded: [/content/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors](https://colab.research.google.com/drive/1xwBjrCWtZvh-XlTIk2lyQ5sFjDxkcsqW#) VAE loaded: None Request to load LoRAs [('sd_xl_offset_example-lora_1.0.safetensors', 0.1)] for model [/content/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors]. Loaded LoRA [/content/Fooocus/models/loras/sd_xl_offset_example-lora_1.0.safetensors] for UNet [/content/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 788 keys at weight 0.1. Fooocus V2 Expansion: Vocab with 642 words. Fooocus Expansion engine loaded for cpu, use_fp16 = False. Requested to load SDXLClipModel Requested to load GPT2LMHeadModel Loading 2 new models Started worker with PID 8748 App started successful. Use the app with [http://127.0.0.1:7865/](https://www.google.com/url?q=http%3A%2F%2F127.0.0.1%3A7865%2F) or 127.0.0.1:7865 or [https://1b6636ec14a73f6262.gradio.live](https://www.google.com/url?q=https%3A%2F%2F1b6636ec14a73f6262.gradio.live) [Parameters] Adaptive CFG = 7 [Parameters] CLIP Skip = 2 [Parameters] Sharpness = 2 [Parameters] ControlNet Softness = 0.25 [Parameters] ADM Scale = 1.5 : 0.8 : 0.3 [Parameters] Seed = 6137604482871942681 [Parameters] CFG = 4 [Fooocus] Loading control models ... [Parameters] Sampler = dpmpp_2m_sde_gpu - karras [Parameters] Steps = 30 - 15 [Fooocus] Initializing ... [Fooocus] Loading models ... Refiner unloaded. [Fooocus] Processing prompts ... [Fooocus] Preparing Fooocus text #1 ... [Prompt Expansion] dog, dramatic, bright colors, sharp focus, intricate, elegant, highly detailed, beautiful, attractive, depicted, rich, meticulous, creative, light novel, fancy, cute, cheerful, glowing, artistic, fine detail, colorful, winning, complex, elaborate, enhanced, color, clear, fair, perfect, pretty, radiant, loving, delicate, friendly, quiet [Fooocus] Preparing Fooocus text #2 ... [Prompt Expansion] dog, cinematic, epic, dynamic light, ambient, crisp quality, highly detailed, professional, fair, clear, color, artistic, winning, complex, imposing, contemporary, rich, vivid, beautiful, infinite, symmetry, great, perfect, atmosphere, amazing, elegant, sharp focus, creative, fantastic, expressive, awesome, full, fabulous, marvelous, glowing, wonderful [Fooocus] Encoding positive #1 ... [Fooocus] Encoding positive #2 ... [Fooocus] Encoding negative #1 ... [Fooocus] Encoding negative #2 ... [Parameters] Denoising Strength = 1.0 [Parameters] Initial Latent shape: Image Space (896, 1152) Preparation time: 6.08 seconds Using karras scheduler. [Fooocus] Preparing task 1/2 ... [Sampler] refiner_swap_method = joint [Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828 Requested to load SDXL Loading 1 new model [Fooocus Model Management] Moving model(s) has taken 5.08 seconds 100% 30/30 [10:53<00:00, 21.79s/it] Requested to load AutoencoderKL Loading 1 new model [Fooocus] Saving image 1/2 to system ... Image generated with private log at: [/content/Fooocus/outputs/2024-10-03/log.html](https://colab.research.google.com/drive/1xwBjrCWtZvh-XlTIk2lyQ5sFjDxkcsqW#) Generating and saving time: 669.16 seconds [Fooocus] Preparing task 2/2 ... [Sampler] refiner_swap_method = joint [Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828 100% 30/30 [09:35<00:00, 19.20s/it] [Fooocus] Saving image 2/2 to system ... Image generated with private log at: [/content/Fooocus/outputs/2024-10-03/log.html](https://colab.research.google.com/drive/1xwBjrCWtZvh-XlTIk2lyQ5sFjDxkcsqW#) Generating and saving time: 585.94 seconds [Enhance] Skipping, preconditions aren't met Processing time (total): 1255.10 seconds Total time: 1261.22 seconds

**Comment by werruww** [comment]
https://colab.research.google.com/drive/1xwBjrCWtZvh-XlTIk2lyQ5sFjDxkcsqW#scrollTo=2QjwpvNpFrRJ

**Comment by werruww** [comment]
![2024-10-03_20-19-30_2172](https://github.com/user-attachments/assets/757af66f-60e5-4397-afec-d64373ffed6c)
![2024-10-03_20-29-16_3481](https://github.com/user-attachments/assets/0b1ccad1-3b26-475d-a8b7-1a6136a6ac2f)
![Web capture_3-10-2024_132448_1b6636ec14a73f6262 gradio live](https://github.com/user-attachments/assets/004cf142-8069-471c-8231-6963c5803575)
![Web capture_3-10-2024_132524_colab research google com](https://github.com/user-attachments/assets/6c731d6c-d073-42eb-8597-e69bd3be000d)
![Web capture_3-10-2024_133411_1b6636ec14a73f6262 gradio live](https://github.com/user-attachments/assets/2f833b9a-599d-4860-a61d-a7f475a677ea)

**Reply by poor7** [reply]
What's the point if you can generate much faster on a GPU?

**Comment by werruww** [comment]
iam use tpu for big ram amd run focuse on cpu

**Comment by werruww** [comment]
Already up-to-date
Update succeeded.
[System ARGV] ['entry_with_update.py', '--share', '--cpu-only']
Python 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0]
Fooocus version: 2.5.5
usage: entry_with_update.py [-h] [--listen [IP]] [--port PORT] [--disable-header-check [ORIGIN]]
                            [--web-upload-size WEB_UPLOAD_SIZE] [--hf-mirror HF_MIRROR]
                            [--external-working-path PATH [PATH ...]] [--output-path OUTPUT_PATH]
                            [--temp-path TEMP_PATH] [--cache-path CACHE_PATH] [--in-browser]
                            [--disable-in-browser] [--gpu-device-id DEVICE_ID]
                            [--async-cuda-allocation | --disable-async-cuda-allocation]
                            [--disable-attention-upcast] [--all-in-fp32 | --all-in-fp16]
                            [--unet-in-bf16 | --unet-in-fp16 | --unet-in-fp8-e4m3fn | --unet-in-fp8-e5m2]
                            [--vae-in-fp16 | --vae-in-fp32 | --vae-in-bf16] [--vae-in-cpu]
                            [--clip-in-fp8-e4m3fn | --clip-in-fp8-e5m2 | --clip-in-fp16 | --clip-in-fp32]
                            [--directml [DIRECTML_DEVICE]] [--disable-ipex-hijack]
                            [--preview-option [none,auto,fast,taesd]]
                            [--attention-split | --attention-quad | --attention-pytorch]
                            [--disable-xformers]
                            [--always-gpu | --always-high-vram | --always-normal-vram | --always-low-vram | --always-no-vram | --always-cpu [CPU_NUM_THREADS]]
                            [--always-offload-from-vram] [--pytorch-deterministic]
                            [--disable-server-log] [--debug-mode] [--is-windows-embedded-python]
                            [--disable-server-info] [--multi-user] [--share] [--preset PRESET]
                            [--disable-preset-selection] [--language LANGUAGE]
                            [--disable-offload-from-vram] [--theme THEME] [--disable-image-log]
                            [--disable-analytics] [--disable-metadata] [--disable-preset-download]
                            [--disable-enhance-output-sorting] [--enable-auto-describe-image]
                            [--always-download-new-model] [--rebuild-hash-cache [CPU_NUM_THREADS]]
entry_with_update.py: error: unrecognized arguments: --cpu-only
